app/views/pieces/show.html.haml

```diff
@@ -17,5 +18,10 @@
         %li= review.title
         %li= review.body
         %li= "written by: #{review.user.username}"
+        - unless current_user.nil?
+          - if current_user == @piece.user || current_user.admin
+            .links
+              %p= link_to "Edit Review", edit_piece_review_path(@piece, review)
+              %p= link_to "Delete Review", piece_review_path(@piece, review), method: :delete
```

Comment:
Good decision to ignore "line is too long" here... Line looks fine despite length